File: web/types/block.ts
export type Block = {
  id: string;
  semantic_type: string;
  content: string;
  state: string;
  scope: string | null;
  canonical_value: string | null;
  actor?: string;
  created_at?: string;
};

export type BlockWithHistory = Block & {
  prev_rev_id?: string | null;
  prev_content?: string | null;
};

File: shared/substrate/models/block.py
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel

block_state = Literal[
    "PROPOSED",
    "ACCEPTED",
    "LOCKED",
    "CONSTANT",
    "SUPERSEDED",
    "REJECTED",
]


class Block(BaseModel):
    id: UUID
    basket_id: UUID | None = None
    parent_block_id: UUID | None = None
    semantic_type: str
    content: str | None = None
    version: int
    state: block_state
    scope: str | None = None
    canonical_value: str | None = None
    origin_ref: UUID | None = None
    created_at: datetime
